client/src/components/Form/FormRepeat.vue

```diff
@@ -97,7 +93,7 @@ const { keyObject } = useKeyedObjects();
 
         <FormCard
             v-for="(cache, cacheId) in props.input.cache"
-            :key="keyObject(cache)"
+            :key="cacheId"
```

Comment:
This key will not track changes in array order.
`cacheId` is a bit of an misleading name, as this is simply an array index. Array indexes work fine for keying static lists, but break on dynamic ones. This was the reason for the object keying.

Reply:
When restoring the reactivity of the tool form, using the `keyObject` tracking prevents the selection of datasets in repeats. However, I am ok with leaving this untouched in this PR since it is unrelated. I have removed the general fixes for the tool form from this PR.

Reply:
I think that bug should be easier to resolve once this PR is merged, without the need for the object deep copy